pkg/api/authn/noauth.go

```diff
@@ -1,20 +1,44 @@
 package authn
 
 import (
+	"fmt"
 	"net/http"
 
+	types2 "github.com/obot-platform/obot/apiclient/types"
 	"github.com/obot-platform/obot/pkg/api/authz"
+	"github.com/obot-platform/obot/pkg/gateway/client"
+	"github.com/obot-platform/obot/pkg/gateway/types"
 	"k8s.io/apiserver/pkg/authentication/authenticator"
 	"k8s.io/apiserver/pkg/authentication/user"
 )
 
 type NoAuth struct {
+	client *client.Client
 }
 
-func (n NoAuth) AuthenticateRequest(*http.Request) (*authenticator.Response, bool, error) {
+func NewNoAuth(client *client.Client) *NoAuth {
+	return &NoAuth{
+		client: client,
+	}
+}
+
+func (n *NoAuth) AuthenticateRequest(req *http.Request) (*authenticator.Response, bool, error) {
+	gatewayUser, err := n.client.EnsureIdentityWithRole(
+		req.Context(),
+		&types.Identity{
+			ProviderUsername: "nobody",
+		},
```

Comment:
We don't want this user in the database.

Reply:
Never mind. This was misguided and Nick has shown me the light.

Reply:
I added some cleanup logic when auth is enabled that ensure we drop this user from the DB on startup. This should reduce the potential attack surface.